ghcide/src/Development/IDE/Spans/AtPoint.hs

```diff
@@ -203,6 +205,51 @@ gotoDefinition
 gotoDefinition withHieDb getHieFile ideOpts imports srcSpans pos
   = lift $ locationsAtPoint withHieDb getHieFile ideOpts imports pos srcSpans
 
+fixityAtPoint :: HieAstResult -> HiFileResult -> HscEnv -> Position -> IO (Maybe (Maybe Range, [T.Text]))
```

Comment:
I think it wouldn't be crazy to make `atPoint` run in `MonadIO m`: almost every other function in here does.

Reply:
A little crazy to make a pure function to impure?😉

The pain point to making fixity query be impure is `runTcInteractive`, I feel it may have a pure method here, worth trying.

Reply:
Sure, that's generally bad, but most everything here has to run in some monad because it's doing stuff with the GHC API. It's a nice coincidence that `atPoint` didn't need it, but I don't think it's fundamental.